### PR TITLE
[reporting] deal with UTF-8 characters

### PR DESCRIPTION
**Pull request by pmoravec**
replace str class functions by unicode variants:
1) "\n".join(buf) needs to pass decoded UTF-8 text
2) fd.write needs to replace str(..) by some other function name
3) fd.write needs to encode UTF-8

Resolves: #722

Signed-off-by: Pavel Moravec <pmoravec@redhat.com>

### CODE REVIEW

**Comment by pmoravec** [comment]
How can I force Travis CI to re-build? I fixed the trivial pep8 error and pushed to this branch.

**Comment by bmr-cymru** [comment]
It should just do it (and it looks like it did..)

If in doubt you can always visit the [sos Travis page](https://travis-ci.org/sosreport/sos/) and poke around the list of recent jobs.

**Comment by pmoravec** [comment]
I have updated it. Though dont like the `if six.PY2:` and nasty `map(lambda ..` bits in `reporting.py`, I am not aware of simplier fix :-/

File: sos/reporting.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
 # PYCOMPAT
 from six import iteritems
+import six
```
Comment (bmr-cymru) [inline sos/reporting.py]: @sbradley7777 be aware of this for RHEL6 (the de-six patch will need updating).

**Comment by bmr-cymru** [comment]
Not a fan of those mapped lambdas either.. I can't help feeling that this shouldn't be as complex as we're making it (fixing the odd unicode problem has spiraled into lots of quite gnarly bits of code in various classes that do IO).

**Comment by battlemidget** [comment]
This branch seems to have been outdated and should be reviewed again to see if the conflicts need to be resolved

**Comment by pmoravec** [comment]
The PR has been fixed in upstream by commit 1fd12690870e85e8ac83b0e99bb272ce4489dc60 thus closing it.

**Pull request closed without merging**